testinfra/modules/mountpoint.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2015 Philippe Pepiot
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+from testinfra.modules.base import Module
+
+
+class MountPoint(Module):
+    """Test Mount Points"""
+
+    def __init__(self, path):
+        self.path = path
+        super(MountPoint, self).__init__()
+
+    def __repr__(self):
+        return "<mountpoint %s>" % (self.path,)
+
+    def find_mountpoint(self):
+        raise NotImplementedError
+
+    @property
+    def filesystem(self):
+        """Returns the FileSystem type associated to a mount point
+
+        >>> MountPoint("/").filesystem
+        'ext4'
+
+        """
+        mountpoint = self.find_mountpoint()
```

Comment:
Instead of calling `find_mountpoint()` on each property. Add a private attribute for caching values.

Something like:

```python
def __init__(self, path):
    self._attrs = None
    self.path = path
    super(MountPoint, self).__init__()

def _get_attrs(self):
    if self._attrs is None:
        self._attrs = self.get_mountpoint()
    return self._attrs

@property
def device(self):
   return self._get_attrs()["device"]
```